backend/src/main/java/ai/verta/modeldb/authservice/RoleServiceUtils.java

```diff
@@ -1148,8 +1150,96 @@ public WorkspaceDTO getWorkspaceDTOByWorkspaceId(
     }
   }
 
+  private boolean createWorkspacePermissions(
```

Comment:
I'm pretty confident I already reviewed this method. Is this being moved from somewhere? If yes, what are the changes? It's really hard for me to re-review the same code again.

Reply:
This is a refactor to support using either the workspaceServiceId or the workspaceName.  Those methods now both call this shared private implementation that performs the UAC call.  This function accepts optional parameters for the workspace ID, name, type, and owner, and populates whatever is provided onto the grpc call to UAC.  

This mehtod is called from https://github.com/VertaAI/modeldb/pull/1750/files#diff-c8d019d1d18aa75e6ffdc3d9eee30bdda569c52ea122ddfbdbe326693460ec56R1190-R1198 and https://github.com/VertaAI/modeldb/pull/1750/files#diff-c8d019d1d18aa75e6ffdc3d9eee30bdda569c52ea122ddfbdbe326693460ec56R1210-R1218 below.

Reply:
Could you highlight what are the differences? I'm pretty sure I'll have a really hard time understanding how it connects to other things due to the size of the method and PR. Having a better idea of what's supposed to have changed will help me focus on that.

Reply:
Sure thing.
https://github.com/VertaAI/modeldb/pull/1750/files#diff-f93a82e7dcaa2d1dbb8859f86aba1d7526b392ac93d4a53896041f414a62d4d2R139-R155 add two new interface methods.  The first is for creating permissions with a `workspaceName`, the second uses the `workspaceServiceId`.

Reply:
In a previous commit, they have nearly identical method bodies.  I consolidated them both to this singular method which takes in both values as Optional and populates them accordingly.

Reply:
https://github.com/VertaAI/modeldb/commit/f077f37e9d078fd0fa5870831e55d55238545031#diff-c8d019d1d18aa75e6ffdc3d9eee30bdda569c52ea122ddfbdbe326693460ec56R1164-R1250

That is the introduction of the refactor, which adds the overload that accepts the workspaceName and then extracts this private method to handle the logic.

Reply:
It all comes down to these lines: https://github.com/VertaAI/modeldb/commit/f077f37e9d078fd0fa5870831e55d55238545031#diff-c8d019d1d18aa75e6ffdc3d9eee30bdda569c52ea122ddfbdbe326693460ec56R1190-R1197

Which check to see if the parameter is provided and add it to the request if so.